components/brave_ads/core/internal/legacy_migration/client/legacy_client_migration.cc

```diff
@@ -48,7 +48,18 @@ void MigrateClientState(InitializeCallback callback) {
              }
 
              ClientInfo client;
-             if (!client.FromJson(*json)) {
+
+             std::string mutable_json = *json;
+             if (!client.FromJson(mutable_json) &&
+                 !GetProfileBooleanPref(
+                     "brave.brave_ads.state.has_migrated.client.v6")) {
```

Comment:
kHasMigratedClientState

Reply:
We can't because if we do they when we change version this test will pass and we need it to fail